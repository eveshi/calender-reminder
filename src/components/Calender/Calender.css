*,
*::after,
*::before {
  margin: 0;
  padding: 0;
  box-sizing: border-box; }

html {
  font-size: 62.5%; }
  @media only screen and (max-width: 400px) {
    html {
      font-size: 50%; } }

body {
  font-family: 'Roboto', sans-serif;
  background-color: #F5F5F4; }

a svg {
  height: 4rem;
  width: 4rem;
  text-decoration: none;
  fill: #5F513F; }
  a svg:hover {
    background-color: transparent; }

@font-face {
  font-family: 'Roboto', sans-serif;
  src: url("https://fonts.googleapis.com/css?family=Roboto"); }

.calender {
  max-width: inherit;
  display: grid;
  grid-template-columns: repeat(7, 1fr);
  justify-items: center;
  background-color: #F5F5F4;
  padding-top: 2rem;
  padding-bottom: 2rem; }
  .calender__weekday {
    height: 2rem;
    font-size: 1.8rem;
    color: #BD924B; }
