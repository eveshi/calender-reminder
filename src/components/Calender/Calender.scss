@import '../../stylesheets/main.scss';

.calender {
  max-width: 600px;
  display: grid;
  grid-template-columns: repeat(7, 1fr);
  align-items: center;
  justify-content: center;

  &__weekday {
    height: 2rem;
  }
}