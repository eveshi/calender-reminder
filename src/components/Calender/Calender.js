import React from 'react';
import PropTypes from 'prop-types';
import dateFns from 'date-fns';
import CalenderCell from '../CalenderCell/CalenderCell';
import './Calender.css';

const Calender = (props) => {
  const {
    year,
    month,
    day,
    onClick,
  } = props;

  const date = new Date(year, month, day);
  const daysOfMonth = dateFns.getDaysInMonth(date);
  const firstDayOfMonth = dateFns.startOfMonth(date);
  const weekdayOfStartDayOfMonth = dateFns.getDay(new Date(firstDayOfMonth));

  let fillArray = [];
  for (let i = 1; i < weekdayOfStartDayOfMonth + 1; i += 1) {
    fillArray.push(i);
  }
  if (weekdayOfStartDayOfMonth === 7) {
    fillArray = [];
  }

  const daysArray = [];
  for (let i = 1; i < daysOfMonth + 1; i += 1) {
    daysArray.push(i);
  }

  const fills = fillArray === null
    ? null
    : fillArray.map(fill => (
      <div key={fill} />
    ));

  const cells = daysArray === null
    ? null
    : daysArray.map((dayItem) => {
      const picked = (dayItem === day);
      return (
        <CalenderCell
          key={dayItem}
          day={dayItem}
          picked={picked}
          onClick={onClick}
        />
      );
    });

  return (
    <div className="calender">
      <p className="calender__weekday">Sun</p>
      <p className="calender__weekday">Mon</p>
      <p className="calender__weekday">Tue</p>
      <p className="calender__weekday">Wed</p>
      <p className="calender__weekday">Thu</p>
      <p className="calender__weekday">Fri</p>
      <p className="calender__weekday">Sat</p>
      {fills}
      {cells}
    </div>
  );
};

Calender.propTypes = {
  year: PropTypes.number.isRequired,
  month: PropTypes.number.isRequired,
  day: PropTypes.number.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default Calender;
