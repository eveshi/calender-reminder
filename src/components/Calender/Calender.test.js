import React from 'react';
import renderer from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Calender from './Calender';

configure({ adapter: new Adapter() });

const props = {
  year: 2033,
  month: 6,
  day: 2,
  onClick: jest.fn,
};

describe('snapshot', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Calender {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
