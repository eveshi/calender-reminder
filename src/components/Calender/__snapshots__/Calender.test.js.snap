// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot renders correctly 1`] = `
<div
  className="calender"
>
  <p
    className="calender__weekday"
  >
    Sun
  </p>
  <p
    className="calender__weekday"
  >
    Mon
  </p>
  <p
    className="calender__weekday"
  >
    Tue
  </p>
  <p
    className="calender__weekday"
  >
    Wed
  </p>
  <p
    className="calender__weekday"
  >
    Thu
  </p>
  <p
    className="calender__weekday"
  >
    Fri
  </p>
  <p
    className="calender__weekday"
  >
    Sat
  </p>
  <div />
  <div />
  <div />
  <div />
  <div />
  <div
    className="Calender_CalenderCell__notPicked"
  >
    <button
      className="Button"
      disabled={null}
      id={1}
      onClick={[Function]}
      type="button"
    >
      1
    </button>
  </div>
  <div
    className="Calender_CalenderCell__picked"
  >
    <button
      className="Button"
      disabled={null}
      id={2}
      onClick={[Function]}
      type="button"
    >
      2
    </button>
  </div>
  <div
    className="Calender_CalenderCell__notPicked"
  >
    <button
      className="Button"
      disabled={null}
      id={3}
      onClick={[Function]}
      type="button"
    >
      3
    </button>
  </div>
  <div
    className="Calender_CalenderCell__notPicked"
  >
    <button
      className="Button"
      disabled={null}
      id={4}
      onClick={[Function]}
      type="button"
    >
      4
    </button>
  </div>
  <div
    className="Calender_CalenderCell__notPicked"
  >
    <button
      className="Button"
      disabled={null}
      id={5}
      onClick={[Function]}
      type="button"
    >
      5
    </button>
  </div>
  <div
    className="Calender_CalenderCell__notPicked"
  >
    <button
      className="Button"
      disabled={null}
      id={6}
      onClick={[Function]}
      type="button"
    >
      6
    </button>
  </div>
  <div
    className="Calender_CalenderCell__notPicked"
  >
    <button
      className="Button"
      disabled={null}
      id={7}
      onClick={[Function]}
      type="button"
    >
      7
    </button>
  </div>
  <div
    className="Calender_CalenderCell__notPicked"
  >
    <button
      className="Button"
      disabled={null}
      id={8}
      onClick={[Function]}
      type="button"
    >
      8
    </button>
  </div>
  <div
    className="Calender_CalenderCell__notPicked"
  >
    <button
      className="Button"
      disabled={null}
      id={9}
      onClick={[Function]}
      type="button"
    >
      9
    </button>
  </div>
  <div
    className="Calender_CalenderCell__notPicked"
  >
    <button
      className="Button"
      disabled={null}
      id={10}
      onClick={[Function]}
      type="button"
    >
      10
    </button>
  </div>
  <div
    className="Calender_CalenderCell__notPicked"
  >
    <button
      className="Button"
      disabled={null}
      id={11}
      onClick={[Function]}
      type="button"
    >
      11
    </button>
  </div>
  <div
    className="Calender_CalenderCell__notPicked"
  >
    <button
      className="Button"
      disabled={null}
      id={12}
      onClick={[Function]}
      type="button"
    >
      12
    </button>
  </div>
  <div
    className="Calender_CalenderCell__notPicked"
  >
    <button
      className="Button"
      disabled={null}
      id={13}
      onClick={[Function]}
      type="button"
    >
      13
    </button>
  </div>
  <div
    className="Calender_CalenderCell__notPicked"
  >
    <button
      className="Button"
      disabled={null}
      id={14}
      onClick={[Function]}
      type="button"
    >
      14
    </button>
  </div>
  <div
    className="Calender_CalenderCell__notPicked"
  >
    <button
      className="Button"
      disabled={null}
      id={15}
      onClick={[Function]}
      type="button"
    >
      15
    </button>
  </div>
  <div
    className="Calender_CalenderCell__notPicked"
  >
    <button
      className="Button"
      disabled={null}
      id={16}
      onClick={[Function]}
      type="button"
    >
      16
    </button>
  </div>
  <div
    className="Calender_CalenderCell__notPicked"
  >
    <button
      className="Button"
      disabled={null}
      id={17}
      onClick={[Function]}
      type="button"
    >
      17
    </button>
  </div>
  <div
    className="Calender_CalenderCell__notPicked"
  >
    <button
      className="Button"
      disabled={null}
      id={18}
      onClick={[Function]}
      type="button"
    >
      18
    </button>
  </div>
  <div
    className="Calender_CalenderCell__notPicked"
  >
    <button
      className="Button"
      disabled={null}
      id={19}
      onClick={[Function]}
      type="button"
    >
      19
    </button>
  </div>
  <div
    className="Calender_CalenderCell__notPicked"
  >
    <button
      className="Button"
      disabled={null}
      id={20}
      onClick={[Function]}
      type="button"
    >
      20
    </button>
  </div>
  <div
    className="Calender_CalenderCell__notPicked"
  >
    <button
      className="Button"
      disabled={null}
      id={21}
      onClick={[Function]}
      type="button"
    >
      21
    </button>
  </div>
  <div
    className="Calender_CalenderCell__notPicked"
  >
    <button
      className="Button"
      disabled={null}
      id={22}
      onClick={[Function]}
      type="button"
    >
      22
    </button>
  </div>
  <div
    className="Calender_CalenderCell__notPicked"
  >
    <button
      className="Button"
      disabled={null}
      id={23}
      onClick={[Function]}
      type="button"
    >
      23
    </button>
  </div>
  <div
    className="Calender_CalenderCell__notPicked"
  >
    <button
      className="Button"
      disabled={null}
      id={24}
      onClick={[Function]}
      type="button"
    >
      24
    </button>
  </div>
  <div
    className="Calender_CalenderCell__notPicked"
  >
    <button
      className="Button"
      disabled={null}
      id={25}
      onClick={[Function]}
      type="button"
    >
      25
    </button>
  </div>
  <div
    className="Calender_CalenderCell__notPicked"
  >
    <button
      className="Button"
      disabled={null}
      id={26}
      onClick={[Function]}
      type="button"
    >
      26
    </button>
  </div>
  <div
    className="Calender_CalenderCell__notPicked"
  >
    <button
      className="Button"
      disabled={null}
      id={27}
      onClick={[Function]}
      type="button"
    >
      27
    </button>
  </div>
  <div
    className="Calender_CalenderCell__notPicked"
  >
    <button
      className="Button"
      disabled={null}
      id={28}
      onClick={[Function]}
      type="button"
    >
      28
    </button>
  </div>
  <div
    className="Calender_CalenderCell__notPicked"
  >
    <button
      className="Button"
      disabled={null}
      id={29}
      onClick={[Function]}
      type="button"
    >
      29
    </button>
  </div>
  <div
    className="Calender_CalenderCell__notPicked"
  >
    <button
      className="Button"
      disabled={null}
      id={30}
      onClick={[Function]}
      type="button"
    >
      30
    </button>
  </div>
  <div
    className="Calender_CalenderCell__notPicked"
  >
    <button
      className="Button"
      disabled={null}
      id={31}
      onClick={[Function]}
      type="button"
    >
      31
    </button>
  </div>
</div>
`;
